src/nvim/window.c

```diff
@@ -5640,28 +5635,28 @@ int match_add(win_T *wp, const char *const grp, const char *const pat,
           continue;
```

Comment:
The iterative statement of the loop does not increase `i` anymore, so decreasing `i` before a `continue` is wrong.